### ISSUE TITLE
`aac_normalize` throws despite `try`

### ISSUE THREAD

**Issue by aa755** [coq-community/aac-tactics#144]
```
Lemma foo: forall X:Prop, X->X.
try aac_normalise.
(* Error: [aac_tactics] The goal is not an applied relation *)
```
I would like to call `try aac_normalise` on all goals but this prevents me from doing so. Is there a workaround?

Also, the exception that `aac_normalise`  throws seems wierd: once it threw when it was called in a deep ltac2 stack, but the ltac2 backtrace was empty. (I can produce a minimal example if that helps.) Does it discard the entire stack? 

Can `aac_normalise` just not fail? A user can check whether it did anything by wrapping it around `progress`. The missing call stack would make it hard to debug failures of complex tactics that call `aac_normalise` if that fails

**Comment by aa755** [comment]
`aac_rewrite` has the same behaviour:
```
Lemma foo: forall X:Prop, X->X.
try aac_rewrite Z.gcd_mod.
(* Error: [aac_tactics] The goal is not an applied relation *)
```

**Comment by aa755** [comment]
I cannot catch the exception even in ltac2:
```
Lemma foo: forall X:Prop, X->X.
  ltac2:(Control.plus (fun _ => ltac1:(aac_rewrite Z.gcd_mod)) (fun _ => ())).
  (* Error: [aac_tactics] The goal is not an applied relation *)
```

**Comment by SkySkimmer** [comment]
Ltac2 is not more capable of catching exceptions than ltac1 so that's normal.

**Comment by SkySkimmer** [comment]
Cannot reproduce with 
~~~coq
From AAC_tactics Require Import AAC.

Lemma foo: forall X:Prop, X->X.
  try aac_normalise.
  (* succeeds doing nothing *)
~~~
on coq master, aac-tactics master

**Comment by aa755** [comment]
Thanks for investigating. I will try to reproduce it on coq master, aac-tactics master and report back.

**Comment by aa755** [comment]
I can confirm that this issue does not occur with coq master.
https://gist.github.com/aa755/a7ae663b196ee60f9ff4c601e76bc976

The implementation of aac tactics seem to have not changed since the version above where I can reproduce the issue. So it seems this got fixed by some changes in coq.

**Comment by palmskog** [comment]
@aa755 could you check if it occurs on Coq's `v8.20` branch? If the issue is gone there, then this bug will be gone when I release AAC Tactics for 8.20, which will happen shortly after the release candidate for Coq 8.20.0 is out (which I think is in a few weeks).

**Comment by palmskog** [comment]
There is already a `v8.20` branch of AAC Tactics, but currently it's identical to `master`: https://github.com/coq-community/aac-tactics/tree/v8.20

**Comment by SkySkimmer** [comment]
8.20 RC is expected tomorrow according to https://github.com/coq/coq/wiki/Release-Schedule-for-Coq-8.20

**Comment by aa755** [comment]
> There is already a `v8.20` branch of AAC Tactics, but currently it's identical to `master`: https://github.com/coq-community/aac-tactics/tree/v8.20

So, there is nothing to test for now? If you want me to test something, let me know and I will be happy to.

**Comment by palmskog** [comment]
@aa755 I can confirm that the following works on Coq 8.20+rc1:

```coq
From AAC_tactics Require Import AAC.
From Coq Require Import ZArith.

Goal forall X:Prop, X->X.
Succeed (try aac_rewrite Z.gcd_mod).
Abort.

Goal forall X:Prop, X->X.
Succeed (try aac_normalise).
Abort.
```
So I will close this issue after I release AAC Tactics 8.20.0 (will happen soon). Please reopen if the problem reappears. Unfortunately, this will not be solved on Coq 8.19.

**Comment by palmskog** [comment]
Release for 8.20 is done and submitted to the Coq opam archive (`extra-dev` repo for the moment, but will be moved to `released` once 8.20.0 is out): https://github.com/coq/opam/pull/3059

**Comment by aa755** [comment]
Thanks. Sorry I was not able to help soon enough. I was planning to do it this weekend.
